frontend/src/components/Card.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import Rating from './Rating';
+import { Link } from 'react-router-dom';
+import { ADD_TO_WISHLIST } from '../constants/WishListConstants';
+import useCustomContext from '../customHooks/Hook';
+
+const Card = ({ product }) => {
+
+    const { state: { wishlist }, dispatch } = useCustomContext();
+    const { id, name, image, price, numReviews, rating } = product;
+
+    const addToWishlist = () => {
+        dispatch({ type: ADD_TO_WISHLIST, payload: product })
+    }
+
+    const check = () => {
+        return !!wishlist.find((x) => x.id === id);
+    }
+
+    return (
+        <div className="product-card">
+
+            <div className="product-image-container">
+                <img src={image} alt='product-image' className="image top-img-card" />
+            </div>
+
+            <div className="product-text-container">
+                <div className="d-flex flex-direction-column m-l-1 flex-grow-1" >
```

Comment:
I have a suggestion, d-flex and m-l-1 can be replaced with more descriptive names

Reply:
done